key-rotator/manifest/manifest.go

```diff
@@ -165,6 +247,52 @@ func validateUpdatedManifest(m, oldM DataShareProcessorSpecificManifest) error {
 	return nil
 }
 
+// validateKeyMaterialAgainstManifest verifies that, for any key versions that
+// exist in both the update config's keys & the manifest's keys, the key
+// material matches. No verification is done for key material that exists in
+// only the update config's keys or only the manifest's keys.
+func validateKeyMaterialAgainstManifest(cfg UpdateKeysConfig, m DataShareProcessorSpecificManifest) error {
+	// Verify batch signing keys.
+	if err := cfg.BatchSigningKey.Versions(func(v key.Version) error {
+		kid := cfg.batchSigningKeyID(v.CreationTimestamp)
+		bsk, ok := m.BatchSigningPublicKeys[kid]
+		if !ok {
+			return nil // key version does not exist in manifest
+		}
+		manifestPubkey, err := bsk.toPublicKey()
+		if err != nil {
+			return fmt.Errorf("couldn't parse batch signing key version %q from manifest: %v", kid, err)
```

Comment:
Nit; was the use of %v on purpose here? They behave a __tiny__ bit differently https://pkg.go.dev/fmt#Errorf

Reply:
(same with a bunch of other areas)

Reply:
Whoops, nope, not intentional. Changed `%v` to `%w` for any errors being passed to `fmt.Errorf`.